Auto code shows msgs & surveys and generate production file

IsaackMwenda opened the pull request:
Hi @as2388  PTAL. Have fixed minor naming issues in pipeline config, upgraded core data module to v0.9.0 and added auto code shows msgs & surveys and generate production file. Pending https://github.com/AfricasVoices/Project-ICRAF/pull/2 https://github.com/AfricasVoices/Project-ICRAF/pull/3 https://github.com/AfricasVoices/Project-ICRAF/pull/4 https://github.com/AfricasVoices/Project-ICRAF/pull/5 https://github.com/AfricasVoices/Project-ICRAF/pull/6 reviews

IsaackMwenda committed:
add docker

IsaackMwenda committed:
Rename practise -> practice

IsaackMwenda committed:
Add drive credential file, rename rapidprodomain

IsaackMwenda committed:
add generate output scripts

IsaackMwenda committed:
Add combine raw datasets, renamed libs -> lib

IsaackMwenda committed:
Add main pipeline module

IsaackMwenda committed:
Delete DockerFile

IsaackMwenda committed:
Create intermmittent dirs,remove g-cloud

IsaackMwenda committed:
Fixed bugs, add drive credential file

IsaackMwenda committed:
Produce coda, production & ICR files

IsaackMwenda committed:
Autocode msgs & surveys,generate coda, prod & ICR files

IsaackMwenda committed:
Upgrade coredata -> v0.9.0

IsaackMwenda committed:
Remane coded_field & coda_filename

IsaackMwenda committed:
Check coda version

IsaackMwenda committed:
Push messages to Coda

IsaackMwenda committed:
Fix import identation

IsaackMwenda committed:
Merge branch 'master' into autoCode_productionFile

IsaackMwenda committed:
Merge branch 'yes_no-scheme' into autoCode_productionFile

IsaackMwenda committed:
Upgrade pipeline infrastructure & coredata module

IsaackMwenda committed:
Fix scheme names,add Drive upload args,

IsaackMwenda committed:
Rename scheme name to match field name

IsaackMwenda committed:
Merge branch 'master' into autoCode_productionFile

IsaackMwenda commented:
@as2388  Due to major changes/conflicts in the project I closed https://github.com/AfricasVoices/Project-ICRAF/pull/6 and addressed all the issues in that PR here. In addition this branch has the autocode messages, generation of production file . PTAL

IsaackMwenda committed:
Add apply manual codes module

as2388 commented on configurations/pipeline_config.json:
```diff
@@ -31,14 +36,20 @@
     {"RapidProKey": "Organizations (Value) - icraf_s01_follow_up_survey", "PipelineKey": "organizations_raw"},
-    {"RapidProKey": "Reasons_For_New_Practices (Value) - icraf_s01_follow_up_survey", "PipelineKey": "new_practice_challenge_raw"},
-    {"RapidProKey": "Upper_Tana_Practices (Value) - icraf_s01_follow_up_survey", "PipelineKey": "upper_tana_practices_reasons_raw"},
+    {"RapidProKey": "Reasons_For_New_Practices (Value) - icraf_s01_follow_up_survey", "PipelineKey": "new_practices_challenge_raw"},
```
Do you mean new_practices_challenge*s*_raw?

(I don't know, just checking!)

IsaackMwenda replied:
Thats right because one may answer more than one challenge. Updated!

as2388 commented on pipeline.py:
```diff
@@ -0,0 +1,218 @@
+          drive_client_wrapper.update_or_create(production_csv_output_path, production_csv_drive_dir,
+                                                  target_file_name=production_csv_drive_file_name,
+                                                  target_folder_is_shared_with_me=True)
+          '''
```
If you're committing commented code you should add a TODO explaining why. 

It's super obvious to me in this case, but it also helps to remind you to uncomment it later.

IsaackMwenda replied:
Nice tip thanks!

as2388 commented on pipeline.py:
```diff
@@ -0,0 +1,218 @@
+                                                  target_folder_is_shared_with_me=True)
+     else:
+          print("Skipping uploading to Google Drive (because the pipeline configuration json does not contain the key "
+                    "'DriveUploadPaths')")
```
```suggestion
                "'DriveUploadPaths')")
```

IsaackMwenda replied:
Fixed!

as2388 commented on src/auto_code_show_messages.py:
```diff
@@ -0,0 +1,72 @@
+    for plan in PipelineConfiguration.RQA_FOLLOW_UP_CODING_PLANS:
+        RQA_FOLLOW_UP_KEYS.append(plan.raw_field)
+
+        SENT_ON_KEY = "sent_on"
```
I think your indentation is wrong here

IsaackMwenda replied:
Nice catch thanks. Updated!

as2388 commented on src/lib/pipeline_configuration.py:
```diff
@@ -57,15 +59,15 @@ def __init__(self, raw_field, coded_field, coda_filename, cleaner=None, code_sch
 
-    PROJECT_START_DATE = isoparse("")
-    PROJECT_END_DATE = isoparse("")
+    PROJECT_START_DATE = isoparse("2019-02-04T00:00:00+0300") #To change before production time
```
Put this comment on the preceding line and start it with "TODO".

Then it becomes easier to see when scanning the file, and you get editor support

as2388 replied:
Might be worth setting the start/end dates to datetime.min/.max to make it more obvious that these dates are temporary

IsaackMwenda replied:
Nice tips. Updated thanks alot.

as2388 commented on src/lib/pipeline_configuration.py:
```diff
@@ -57,15 +59,15 @@ def __init__(self, raw_field, coded_field, coda_filename, cleaner=None, code_sch
 
     # Radio and follow up questions coding plans
-    RQA_FA_CODING_PLANS = [
+    RQA_FOLLOW_UP_CODING_PLANS = [
```
Why are these called follow-up coding plans? Aren't these the main radio question flows?

(Previous projects that I have worked on have had rqa flows which handle the answers to each week's radio question, demographic profiling flows which run after we receive an rqa answer, and then follow-up surveys, which are a bit like demographics but ask follow-on questions about the radio shows, and are sent at set times after the rqa flows. I think I'm confused as to whether you're using "follow-up" in this usual AVF sense or in some new sense unique to this project)

IsaackMwenda replied:
Your right, Its the same context as previous projects. Have combined RQA and Follow up coding plans in one list so that I can apply some of the autocode and apply manual code methods on them without having to rewrite the methods twice. Will give the coding plan a more clear name RQA_&_FOLLOW_UP_CODING_PLANS

as2388 replied:
Is this definitely the right place to put the follow up surveys (the other option being in the same list as the demographics)?

I think it depends on where you want to put that data in the individuals analysis csv i.e. do you have a new row per answer like the RQAs, or fill in a new column for every RQA response, like we do with the demographics? It probably also depends on whether you're expecting one response to each follow-up survey or whether people can send repeat replies to the same question. What does RDA need for this project?

IsaackMwenda replied:
Discussed offline.

as2388 commented on src/lib/channels.py:
```diff
@@ -0,0 +1,116 @@
+
+from src.lib import PipelineConfiguration
+
+class Channels(object):
```
Can you submit this file as a separate PR? This is so we can additionally have someone from RDA as a reviewer of the timestamps

IsaackMwenda replied:
Okay will do thanks

as2388 replied:
Remove this file from here before merging or, if it breaks the pipeline, delete the timestamps.

(This is because I want the timestamps to be introduced in PR https://github.com/AfricasVoices/Project-ICRAF/pull/11)

IsaackMwenda replied:
File deleted

as2388 commented on src/lib/channels.py:
```diff
@@ -0,0 +1,116 @@
+
+    # Time ranges expressed in format (start_of_range_inclusive, end_of_range_exclusive)
+    RADIO_PROMO_RANGES = [
+        ("2019-04-02T00:00:00+0300", "2019-04-04T23:59:59+0300"),
```
If you do this then you're ignoring messages received at e.g. 2019-04-04T23:59:59.500+03:00.
The last end date is exclusive, so instead you can say 2019-04-04T24:00:00+03:00, which means anything before the end of day on April 4th.

IsaackMwenda replied:
Great thanks for this.

IsaackMwenda replied:
Have updated to ("2019-04-02T00:00:00+0300", "2019-04-05T07:00:00+0300"), To cover promos that may be aired early morning before the show.

IsaackMwenda committed:
Rename new_practices_challenges, add follow up run ids

Rename new_practices_challenges, add follow up run ids, rearrange dictionary items

IsaackMwenda committed:
Fix identation and code comment

Fix identation and code comment

IsaackMwenda committed:
Fix identation, Rename rqa_and_follow_up_messages

Fix identation, Rename rqa_and_follow_up_messages

IsaackMwenda committed:
Fix radio promo & show ranges

Fix radio promo & show ranges

IsaackMwenda committed:
Fix comments, Rename RQA_AND_FOLLOW_UP_CODING_PLANS,new_practices_challenges_raw

Fix comments, Rename RQA_AND_FOLLOW_UP_CODING_PLANS,new_practices_challenges_raw

IsaackMwenda committed:
Separated Follow Up coding plans & RQA coding plan

as2388 commented on docker-run.sh:
```diff
@@ -0,0 +1,135 @@
+            PROFILE_CPU=true
+            CPU_PROFILE_OUTPUT_PATH="$2"
+            shift 2;;
+        --drive-upload)
```
Low priority, but this flag isn't needed anymore now that this is defined in pipeline_config.json. I don't mind you merging with it still included, provided you file an issue against the repo to remove it.

IsaackMwenda replied:
Fixed in future pr

as2388 commented on pipeline.py:
```diff
@@ -0,0 +1,220 @@
+     production_csv_output_path = args.production_csv_output_path
+
+     message_paths = [s01e01_input_path, s01e02_input_path, s01e03_input_path, s01e04_input_path, 
+                         s01e05_input_path, s01e06_input_path]
```
```suggestion
                     s01e05_input_path, s01e06_input_path]
```

as2388 commented on src/apply_manual_codes.py:
```diff
@@ -0,0 +1,140 @@
+    @classmethod
+    def apply_manual_codes(cls, user, data, coda_input_dir):
+        # Merge manually coded radio show files into the cleaned dataset
+        for plan in PipelineConfiguration.RQA_FOLLOW_UP_CODING_PLANS:
```
This was changed to RQA + FOLLOW_UP in a future PR right?

If not, don't merge this.

IsaackMwenda replied:
Erroneously committed here. Fixed in future pr

IsaackMwenda committed:
Remove channels & apply_manual_codes, fix identations
